controller/node_controller.go

```diff
@@ -558,6 +561,11 @@ func (nc *NodeController) syncNode(key string) (err error) {
 		}
 	}
 
+	// check if environment settings meet the requirements on current node
+	if err := nc.environmentCheck(kubeNode, node); err != nil {
+		return err
+	}
+
```

Comment:
The error will hinder the remaining tasks. If we upate the error to conditions, we don't need to error out.

Reply:
Removed returning errors. Thanks